### PR TITLE
[2.1.3] Update AddSignalRCore to respect user registered services (#2434)

### PR DESCRIPTION
**Pull request by BrennanConroy**
https://github.com/aspnet/SignalR/issues/2561
Porting change from release/2.2

### CODE REVIEW

**Comment by mikaelm12** [comment]
You add a test be sure to also add it to all appropriate branches

**Comment by BrennanConroy** [comment]
Test added

**Comment by BrennanConroy** [comment]
Looking for approval again now that tests are in

File: test/Microsoft.AspNetCore.SignalR.Tests/AddSignalRTests.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            Assert.IsType<CustomHubActivator<CustomHub>>(serviceProvider.GetRequiredService<IHubActivator<CustomHub>>());
+            Assert.IsType<CustomHubContext<CustomHub>>(serviceProvider.GetRequiredService<IHubContext<CustomHub>>());
+        }
+    }
+
+    public class CustomHub : Hub
+    {
+    }
+
+    public class CustomTHub : Hub<string>
```
Comment (mikaelm12) [inline test/Microsoft.AspNetCore.SignalR.Tests/AddSignalRTests.cs]: Hmm, should there be a another test for `DynamicHub ` here as well?

**Pull request merged**